From b83c2e7ac73cc1e4b3ee954ff5c6cdbda6c41d8e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 27 Feb 2013 20:57:01 +0000
Subject: [PATCH] Resolves: fdo#61241 force area page to size itself for max
 visible controls

Change-Id: I2d67ce71528786ef300ab83ee1e5b1cad6b4abe8
(cherry picked from commit fc8022431a1684f3cfb1e2f7e4ec923a8e71df94)
Reviewed-on: https://gerrit.libreoffice.org/2451
Reviewed-by: Miklos Vajna <vmiklos@suse.cz>
Reviewed-by: Eike Rathke <erack@redhat.com>
Reviewed-by: Muthu Subramanian K <muthusuba@gmail.com>
Tested-by: Muthu Subramanian K <muthusuba@gmail.com>
---
 cui/source/tabpages/tparea.cxx | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/cui/source/tabpages/tparea.cxx b/cui/source/tabpages/tparea.cxx
index 87b410f..47f4878 100644
--- a/cui/source/tabpages/tparea.cxx
+++ b/cui/source/tabpages/tparea.cxx
@@ -765,6 +765,13 @@ SvxAreaTabPage::SvxAreaTabPage( Window* pParent, const SfxItemSet& rInAttrs ) :
     aLbColor.SetAccessibleRelationMemberOf( &aFlProp );
     aMtrFldOffset.SetAccessibleRelationLabeledBy(&aFlOffset);
     aMtrFldOffset.SetAccessibleName(aFlOffset.GetText());
+
+    //fdo#61241 lock down size of this tab page until it's
+    //converted to .ui (remember to use some sizegroups
+    //that take into account hidden frames)
+    Size aSize(GetSizePixel());
+    set_width_request(aSize.Width());
+    set_height_request(aSize.Height());
 }
 
 // -----------------------------------------------------------------------
-- 
1.8.1.2

